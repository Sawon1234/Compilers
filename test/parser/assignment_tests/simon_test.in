// DOES NOT CURRENTLY WORK
class Fib (v : Int) extends InterviewQuestions {
    super();

    fun fib(n : Int) : Int {

        first := 1;
        second := 1;
        while(n > 2) {
            tmp := second;
            second := first + second;
            first := tmp;
            n := n - 1;
        }
        return second;
    }
}

return 5;