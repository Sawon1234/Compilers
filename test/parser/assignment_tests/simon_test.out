class Fib ( v : Int ) extends InterviewQuestions < > { super ( ) fun fib < > ( n : Int ) : Int { if ( n == 1 | n == 2 ) { returns 1 ; } first := 1 ; second := 1 ; while ( n . lessThan < > ( 2 , true ) ) { tmp := second ; second := first . plus < > ( second ) ; first := tmp ; n := n . minus < > ( 1 ) ; } return second ; } }